strapi/strapi
# [Data Transfer] fix transfer transaction close

### What does it do?

- ensures transactions are always closed (rolled back) even on interrupted connections
- ensures errors thrown during processing a ws message are always returned to the the client instead of thrown on the server resulting in the transfer hanging

### Why is it needed?

see above

### How to test it?

Try interrupting transfers (such as with CTRL+C), especially several times in a row with a limited connection pool, and you should always be able to use your admin UI or initiate another transfer afterwards, instead of eventually getting locked out.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
